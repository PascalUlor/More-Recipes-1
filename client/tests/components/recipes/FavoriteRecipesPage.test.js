/* eslint-disable */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import {
  FavoriteRecipesPage
} from '../../../components/FavoriteRecipesPage.jsx';


/**
 * @param { boolean } loading
 *
 * @return { * } null
 */

const props = {
  fetchFavoriteRecipes: jest.fn(),
  deleteFavoriteRecipe: jest.fn(),
  setCurrentRecipe: jest.fn(),
  favoriteRecipes: [],
  isFetching: false,
  deleteSuccess: '',
  deleteError: '',
  currentRecipeId: 1,
  paginationDetails: {}
};

sinon.spy(FavoriteRecipesPage.prototype, 'componentDidMount');
describe('FavoriteRecipesPage component', () => {
  const store = mockStore({});
  const shallowWrapper = shallow((
    <Provider store={store}>
      <Router><FavoriteRecipesPage {...props}/></Router>
    </Provider>
  ));
  it('should render correctly', () => {
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
  it('should work call handlePageChange()', () => {
    sinon.spy(FavoriteRecipesPage.prototype, 'handlePageChange');
    const newProps = {
      ...props,
      isFetching: true
    };
    const shallowWrapper = shallow((
      <Provider store={store}>
        <Router><FavoriteRecipesPage {...newProps}/></Router>
      </Provider>
    ));
  });
});
