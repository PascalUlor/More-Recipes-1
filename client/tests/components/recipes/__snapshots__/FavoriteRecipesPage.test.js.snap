// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteRecipesPage component should render correctly 1`] = `
<BrowserRouter>
  <FavoriteRecipesPage
    currentRecipeId={1}
    deleteError=""
    deleteFavoriteRecipe={[MockFunction]}
    deleteSuccess=""
    favoriteRecipes={Array []}
    fetchFavoriteRecipes={[MockFunction]}
    isFetching={false}
    paginationDetails={Object {}}
    setCurrentRecipe={[MockFunction]}
  />
</BrowserRouter>
`;
